### PR TITLE
Æsthetic improvements and two fixes

### PR DESCRIPTION
**Pull request by Porges**
Fixes: #26, also returns once we hit an error in the loop instead of continuing.

(Mostly) stylistic changes:
- Remove a few unneeded buffers and variables
- `sprintf` → `sprintf_s`
- Simplify `DWORD` extraction code (this needs review – is it possible to store non-DWORD data in DWORD entries like you can shove anything into REG_SZ?)
- Use `nullptr`
- Junk `TCHAR` and its ilk

Needed:
- Need to add a test for dodgy non-null-terminated data, but I'm not sure the best way to actually get this _into_ the registry in the first place, since REG.EXE is going to do it correctly. Can we add adjunct C++ programs to the test code somehow?

### CODE REVIEW

**Comment by shiftkey** [comment]
> Can we add adjunct C++ programs to the test code somehow?

If there's a test harness that's easy enough to setup and run from tasks in `package.json` (or shell out to a script) then let's get this stuff in.

File: src/main.cc [hunk 1]
```diff
@@ -130,8 +129,9 @@ v8::Local<v8::Array> EnumerateValues(HKEY hCurrentKey, Isolate *isolate) {
     else
     {
       char errorMessage[50]; // 39 for message + 10 for int  + 1 for null
```
Comment (shiftkey) [inline src/main.cc]: While we're tidying up comments: `s/null/nul/`

File: src/main.cc [hunk 2]
```diff
@@ -183,7 +183,7 @@ NAN_METHOD(ReadValues)
   else
   {
     char errorMessage[46]; // 35 for message + 10 for int + 1 for null
```
Comment (shiftkey) [inline src/main.cc]: While we're tidying up comments: `s/null/nul/`
